patches/content-browser-BUILD.gn.patch

```diff
@@ -1,13 +1,15 @@
 diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
-index c89b65d27a7e0d639f9cc1468aca4083303860a3..cf3a9cfa10c608f1d7493d34206d34d206a3710a 100644
+index c89b65d27a7e0d639f9cc1468aca4083303860a3..5db63fc03ccf7f1f8d1d72743f23c537f377fa78 100644
 --- a/content/browser/BUILD.gn
 +++ b/content/browser/BUILD.gn
-@@ -2122,6 +2122,12 @@ jumbo_source_set("browser") {
+@@ -2122,6 +2122,14 @@ jumbo_source_set("browser") {
        "//ppapi/proxy:ipc",
        "//ppapi/shared_impl",
      ]
 +    if (brave_chromium_build) {
 +      sources += [
```

Comment:
having a dependency on brave for content is definitely an issue. Dependencies should only go the other way. I believe we can add the request interceptors with a Profile subclass. We already have a subclass of the profile manager

Reply:
Technically speaking, this is not a 'dependency' in terms of building. I just don't see other ways to add sources to content module. And strategically it's better to have this possibility (maybe add brave_content dir? don't know yet), because it seems unlikely we are able to avoid touching content. Same applies for adding files to blink and other dirs (e.g. https://github.com/brave/brave-core/pull/1033/files#diff-481577dd75f1cb6943cb7b0601543491R9)

Also, there is no way to avoid patching - `ProfileManager` doesn't help here, probably we may want to subclass `ProfileIOData`, but this will require adding significantly more ugly code (two subclasses and patching at least two instantiation sites).

Finally, after digging into our future migration to NetworkService, I discovered that the whole infrastructure of `UrlRequestInterceptors` (together with `ChromeNetworkDelegate`) will soon be gone. So I'd suggest to merge this as is, and see how this works in beta. After that, decide if want have this in stable and occasionally migrate this to NetworkService.

Reply:
ProfileManager is what you would use if we wanted a subclass of Profile with our own request interceptors added. See BrowserContext::CreateRequestContext/BrowserContext::CreateRequestContextForStoragePartition
We would add to `content::URLRequestInterceptorScopedVector request_interceptors` and that would get passed to `io_data_.CreateMainRequestContextGetter` in the superclass

Reply:
I don't think we should be patching to add request interceptors and we're going to need a BraveProfile eventually anyway

Reply:
Correct me I'm wrong:
 1) We need to subclass both `ProfileImpl` and `OffTheRecordProfileImpl`
 2) We need to patch instantiation sites to instantiate our subclasses. These are `Profile::CreateProfile` and `Profile::CreateOffTheRecordProfile`

Reply:
Yes, we need to subclass both, but I don't think we need to patch for either one. `Profile::CreateProfile` is called from ProfileManager so we can just call `BraveProfile::CreateProfile` instead in BraveProfileManager. For `CreateOffTheRecordProfile` I think we can do a chromium_src override in profile_impl.cc.  As I said we are going to need to do this eventually anyway for things like session tabs/windows and also if we did have to patch for either of them I'd rather patch for something that is general purpose and I'd also prefer to patch src/chrome over src/content

Reply:
there has also been a lot of user activity in the session tab/window issue and it just got bumped up in priority so I expect we'll be working on it soon

Reply:
I'd even be ok if you patched ProfileIOData for now and created a follow-up issue for the profile. At least then we'd be patching src/chrome where we already have dependencies